bridges/FurAffinityUserBridge.php

```diff
@@ -0,0 +1,109 @@
+<?php
+class FurAffinityUserBridge extends BridgeAbstract {
+	const NAME = 'FurAffinity User Gallery';
+	const URI = 'https://www.furaffinity.net';
+	const MAINTAINER = 'CyberJacob';
+	const PARAMETERS = array(
+		array(
+			'searchUsername' => array(
+				'name' => 'Search Username',
+				'type' => 'text',
+				'required' => true,
+				'title' => 'Username to fetch the gallery for'
+			),
+			'loginUsername' => array(
+				'name' => 'Login Username',
+				'type' => 'text',
+				'required' => true
+			),
+			'loginPassword' => array(
+				'name' => 'Login Password',
+				'type' => 'text',
+				'required' => true
+			)
+		)
+	);
+
+	public function collectData() {
+		$cookies = self::login();
+		$url = self::URI . '/gallery/' . $this->getInput('searchUsername');
+
+		$html = getSimpleHTMLDOM($url, $cookies)
+			or returnServerError('Could not load the user\'s galary page.');
+
+		$submissions = $html->find('section[id=gallery-gallery]', 0)->find('figure');
+		foreach($submissions as $submission) {
+			$item = array();
+			$item['title'] = $submission->find('figcaption', 0)->find('a', 0)->plaintext;
+
+			$thumbnail = $submission->find('a', 0);
+			$thumbnail->href = self::URI . $thumbnail->href;
+
+			$item['content'] = $submission->find('a', 0);
+
+			$this->items[] = $item;
+		}
+	}
+
+	public function getName() {
+		return self::NAME . ' for ' . $this->getInput('searchUsername');
+	}
+
+	public function getURI() {
+		return self::URI . '/user/' . $this->getInput('searchUsername');
+	}
+
+	private function login() {
+		$ch = curl_init(self::URI . '/login/');
+		curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+		curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
+
+		curl_setopt($ch, CURLOPT_USERAGENT, ini_get('user_agent'));
+		curl_setopt($ch, CURLOPT_ENCODING, '');
+		curl_setopt($ch, CURLOPT_PROTOCOLS, CURLPROTO_HTTP | CURLPROTO_HTTPS);
+
+		$fields =
+			'action=login&' .
+			'retard_protection=1&' .
+			'name=' . urlencode($this->getInput('loginUsername')) . '&' .
+			'pass=' . urlencode($this->getInput('loginPassword')) . '&' .
+			'login=Login to Furaffinity';
```

Comment:
Thanks for your contribution!

Travis is currently choking on this part.
Here is a possible fix (not tested):

```php
$fields = implode('&', array(
	'action=login',
	'retard_protection=1',
	'name=' . urlencode($this->getInput('loginUsername')),
	'pass=' . urlencode($this->getInput('loginPassword')),
	'login=Login to Faraffinity'
));
```

Reply:
Thanks, appears to work fine so I've added it to the pull request